examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientS2A.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2015 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworldtls;
+
+import io.grpc.ManagedChannel;
+import io.grpc.StatusRuntimeException;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyChannelBuilder;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.net.ssl.SSLException;
+
+// TODO: change flags to use com.google.common.flags
+
+/**
+ * Requests a greeting from the server with TLS 1.3.
+ */
+public class HelloWorldClientS2A {
+    private static final Logger logger = Logger.getLogger(HelloWorldClientS2A.class.getName());
+    private final ManagedChannel channel;
+    private GreeterGrpc.GreeterBlockingStub blockingStub;
+
+    private static SslContext buildSslContext(String clientCertChainFilePath,
+                                              String clientPrivateKeyFilePath,
+                                              String trustCertCollectionFilePath) throws SSLException {
+        return GrpcSslContexts.forClient()
+            .trustManager(new File(trustCertCollectionFilePath))
+            .keyManager(new File(clientCertChainFilePath), new File(clientPrivateKeyFilePath))
+            .protocols("TLSv1.3")
+            .build();
+    }
+
+    /**
+     * Construct client for accessing RouteGuide server using the existing channel.
+     */
+    HelloWorldClientS2A(ManagedChannel channel) {
+        this.channel = channel;
+        blockingStub = GreeterGrpc.newBlockingStub(channel);
```

Comment:
Can you also inject the stub? I.e. make another parameter to the constructor and pass in `GreeterGrpc.newBlockingStub(channel)`?